goroutine - suddenly crash

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

A bug happened!

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/17345221/cc242a66-3158-4e94-a816-5ab4151b66cb)

```bash
goroutine 39 [running]:
github.com/jandedobbeleer/oh-my-posh/src/platform.(*SegmentsCache).Contains(0x1?, {0xc00079575a?, 0x1?})
        /home/runner/work/oh-my-posh/oh-my-posh/src/platform/shell.go:172 +0x31
reflect.Value.call({0x15ff500?, 0xc00037c3e8?, 0x0?}, {0x16c9dcd, 0x4}, {0xc0000089d8, 0x1, 0xded6ad?})
        /opt/hostedtoolcache/go/src/reflect/value.go:586 +0xb0b
reflect.Value.Call({0x15ff500?, 0xc00037c3e8?, 0x1838728?}, {0xc0000089d8?, 0xc0005921b0?, 0x16c59c0?})
        /opt/hostedtoolcache/go/src/reflect/value.go:370 +0xbc
text/template.safeCall({0x15ff500?, 0xc00037c3e8?, 0xc0005921b0?}, {0xc0000089d8?, 0x183f810?, 0x15db520?})
        /opt/hostedtoolcache/go/src/text/template/funcs.go:367 +0x9d
text/template.(*state).evalCall(0xc0005f1cb0, {0x1688de0?, 0xc0005921b0?, 0x1?}, {0x15ff500?, 0xc00037c3e8?, 0xc0005f1578?}, 0x0, {0x18384e8, 0xc000592120}, ...)
        /opt/hostedtoolcache/go/src/text/template/exec.go:795 +0x868
text/template.(*state).evalField(0xc0005f1cb0, {0x1688de0?, 0xc0005921b0?, 0xdf6f0a?}, {0xc00050c00a, 0x8}, {0x18384e8, 0xc000592120}, {0xc000506080, 0x2, ...}, ...)
        /opt/hostedtoolcache/go/src/text/template/exec.go:644 +0xa1c
text/template.(*state).evalFieldChain(0xf6c7e2?, {0x1688de0?, 0xc0005921b0?, 0xc0005f1858?}, {0x1688de0?, 0xc0005921b0?, 0xc000055900?}, {0x18384e8, 0xc000592120}, {0xc000506060, ...}, ...)
        /opt/hostedtoolcache/go/src/text/template/exec.go:605 +0x1eb
text/template.(*state).evalFieldNode(0xc0005f18f8?, {0x1688de0?, 0xc0005921b0?, 0xc0005e8060?}, 0xc0005f18d0?, {0xc000506080?, 0x35?, 0xe01b47?}, {0x1613740, 0x1e0a690, ...})
        /opt/hostedtoolcache/go/src/text/template/exec.go:569 +0xce
text/template.(*state).evalCommand(0xc0005f1cb0, {0x1688de0?, 0xc0005921b0?, 0xdf6310?}, 0x27?, {0x1613740?, 0x1e0a690?, 0xc0005f19c0?})
        /opt/hostedtoolcache/go/src/text/template/exec.go:497 +0xc9
text/template.(*state).evalPipeline(0xc0005f1cb0, {0x1688de0?, 0xc0005921b0?, 0x1?}, 0xc00009c180)
        /opt/hostedtoolcache/go/src/text/template/exec.go:471 +0x145
text/template.(*state).walkIfOrWith(0xc0005f1cb0, 0xa, {0x1688de0?, 0xc0005921b0?, 0x1662960?}, 0x1?, 0x7b00000000000002?, 0x0)
        /opt/hostedtoolcache/go/src/text/template/exec.go:301 +0xd4
text/template.(*state).walk(0xc0005f1cb0, {0x1688de0?, 0xc0005921b0?, 0x30?}, {0x1838578?, 0xc0005f2080?})
        /opt/hostedtoolcache/go/src/text/template/exec.go:277 +0x173
text/template.(*state).walk(0xc0005f1cb0, {0x1688de0?, 0xc0005921b0?, 0xc0005e8000?}, {0x18385c0?, 0xc000592090?})
        /opt/hostedtoolcache/go/src/text/template/exec.go:280 +0x2d5
text/template.(*Template).execute(0xc000376000, {0x182fe60?, 0xc0005921e0?}, {0x1688de0?, 0xc0005921b0?})
        /opt/hostedtoolcache/go/src/text/template/exec.go:224 +0x2c5
text/template.(*Template).Execute(...)
        /opt/hostedtoolcache/go/src/text/template/exec.go:207
github.com/jandedobbeleer/oh-my-posh/src/template.(*Text).Render(0xc0005f1e58)
        /home/runner/work/oh-my-posh/oh-my-posh/src/template/text.go:83 +0x2e7
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).string(0xc00027d520)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:453 +0x1b8
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).SetText(0xc00027d520)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:509 +0x45
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).setSegmentsText.func1(0x0?)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:126 +0x54
created by github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).setSegmentsText
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:124 +0x98

goroutine 1 [semacquire]:
sync.runtime_Semacquire(0xe22a71?)
        /opt/hostedtoolcache/go/src/runtime/sema.go:62 +0x27
sync.(*WaitGroup).Wait(0x1561878?)
        /opt/hostedtoolcache/go/src/sync/waitgroup.go:116 +0x4b
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).setSegmentsText(0xc0000e8b00)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:129 +0x16b
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).executeSegmentLogic(0xc0000e8b00)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:87 +0x87
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).Init(...)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:65
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Engine).renderBlock(0xc00027acb0, 0xc0000e8b00, 0x0)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/engine.go:155 +0x156
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Engine).Primary(0xc00027acb0)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/prompt.go:37 +0x3b5
github.com/jandedobbeleer/oh-my-posh/src/cli.glob..func14(0x1d8d620?, {0xc0003719a0, 0x1?, 0xa?})
        /home/runner/work/oh-my-posh/oh-my-posh/src/cli/print.go:76 +0x416
github.com/spf13/cobra.(*Command).execute(0x1d8d620, {0xc000371860, 0xa, 0xa})
        /opt/hostedtoolcache/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0x1d8d900)
        /opt/hostedtoolcache/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /opt/hostedtoolcache/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/jandedobbeleer/oh-my-posh/src/cli.Execute()
        /home/runner/work/oh-my-posh/oh-my-posh/src/cli/root.go:38 +0x25
main.main()
        /home/runner/work/oh-my-posh/oh-my-posh/src/main.go:6 +0x17

goroutine 12 [syscall, locked to thread]:
syscall.SyscallN(0x16dc8ba?, {0xc000403d28?, 0x1e0bde0?, 0x0?})
        /opt/hostedtoolcache/go/src/runtime/syscall_windows.go:557 +0x109
syscall.Syscall6(0xc000403db0?, 0xded6fe?, 0x188e5bd0dc8?, 0xc000403db0?, 0xe3fbfa?, 0x0?, 0xc000403db0?, 0x114b1a6?)
        /opt/hostedtoolcache/go/src/runtime/syscall_windows.go:501 +0x50
golang.org/x/sys/windows.(*Proc).Call(0xc0000fb800?, {0xc00050c000?, 0x15e0c40?, 0x1?})
        /opt/hostedtoolcache/go/pkg/mod/golang.org/x/sys@v0.10.0/windows/dll_windows.go:176 +0x1ca
golang.org/x/sys/windows.(*LazyProc).Call(0xc0000fb800, {0xc00050c000, 0x4, 0x4})
        /opt/hostedtoolcache/go/pkg/mod/golang.org/x/sys@v0.10.0/windows/dll_windows.go:348 +0x50
github.com/shirou/gopsutil/v3/internal/common.NewWin32PerformanceCounter({0x16de6ba, 0x16}, {0x16e8295, 0x1e})
        /opt/hostedtoolcache/go/pkg/mod/github.com/shirou/gopsutil/v3@v3.23.6/internal/common/common_windows.go:171 +0x168
github.com/shirou/gopsutil/v3/internal/common.ProcessorQueueLengthCounter(...)
        /opt/hostedtoolcache/go/pkg/mod/github.com/shirou/gopsutil/v3@v3.23.6/internal/common/common_windows.go:196
github.com/shirou/gopsutil/v3/load.loadAvgGoroutine({0x18366b0, 0xc0000341c0})
        /opt/hostedtoolcache/go/pkg/mod/github.com/shirou/gopsutil/v3@v3.23.6/load/load_windows.go:38 +0xec
created by github.com/shirou/gopsutil/v3/load.AvgWithContext.func1
        /opt/hostedtoolcache/go/pkg/mod/github.com/shirou/gopsutil/v3@v3.23.6/load/load_windows.go:74 +0x6c

goroutine 45 [runnable]:
text/template.addFuncs(...)
        /opt/hostedtoolcache/go/src/text/template/funcs.go:107
text/template.(*Template).Funcs(0xc00009a000, 0x4d?)
        /opt/hostedtoolcache/go/src/text/template/template.go:182 +0x20d
github.com/jandedobbeleer/oh-my-posh/src/template.(*Text).Render(0xc00075be58)
        /home/runner/work/oh-my-posh/oh-my-posh/src/template/text.go:74 +0x232
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).string(0xc0003dc000)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:453 +0x1b8
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).SetText(0xc0003dc000)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:509 +0x45
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).setSegmentsText.func1(0x0?)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:126 +0x54
created by github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).setSegmentsText
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:124 +0x98

goroutine 46 [runnable]:
text/template.addFuncs(...)
        /opt/hostedtoolcache/go/src/text/template/funcs.go:107
text/template.(*Template).Funcs(0xc0006a0180, 0x9b?)
        /opt/hostedtoolcache/go/src/text/template/template.go:182 +0x20d
github.com/jandedobbeleer/oh-my-posh/src/template.(*Text).Render(0xc00075de58)
        /home/runner/work/oh-my-posh/oh-my-posh/src/template/text.go:74 +0x232
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).string(0xc0003dc1a0)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:453 +0x1b8
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).SetText(0xc0003dc1a0)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:509 +0x45
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).setSegmentsText.func1(0x0?)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:126 +0x54
created by github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).setSegmentsText
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:124 +0x98
PS>
```


### Theme

[my theme is hosted on github](https://github.com/nklowns/nklowns/blob/bcac6eec3a2eac469f1d670bf0ba39e00f3d7d14/assets/nks.omp.json)

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
PS>oh-my-posh debug --plain

Version: 18.2.0

Shell: pwsh (7.3.6)

Segments:

ConsoleTitle(true)                         -   0 ms - GENESIS-NKS ➔ cloud
os(true)                                   -   1 ms -  ⠀
session(false)                             -   0 ms -
shell(true)                                -   0 ms -   pwsh
sysinfo(true)                              -   0 ms - ⠀ CPU: 93%
sysinfo(true)                              -   0 ms - ⠀󰍛 MEM: 15/16GB
executiontime(false)                       -   0 ms -
git(false)                                 -   5 ms -
text(false)                                -   0 ms -
python(false)                              -   0 ms -
node(false)                                -   0 ms -
java(false)                                -   0 ms -
php(false)                                 -   0 ms -
text(false)                                -   0 ms -
battery(true)                              -   7 ms -   100%
connection(true)                           -   4 ms - ⠀󰖩 ⠀
kubectl(false)                             -  64 ms -
aws(false)                                 -   0 ms -
az(false)                                  -   0 ms -
cftarget(false)                            -   5 ms -
text(true)                                 -   0 ms - ╭─
time(true)                                 -   0 ms -  ♥ 13:29 |
root(false)                                -   0 ms -
path(true)                                 -   1 ms -  󰋜
exit(true)                                 -   0 ms - ╰─

Run duration: 102.6213ms

Cache path: C:\Users\cloud\AppData\Local\oh-my-posh

Config path: C:\Users\cloud\Documents\PowerShell\nks.omp.json

Logs:

[DEBUG] 13:29:41.439 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.439 shell.go:Getenv(TERM_PROGRAM) - 1.236ms
[DEBUG] 13:29:41.439 shell.go:Getenv:398 → C:\Users\cloud\AppData\Local
[TRACE] 13:29:41.439 shell.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 13:29:41.439 shell_windows.go:CachePath() - 0s
[DEBUG] 13:29:41.439 shell.go:Getenv:398 → C:\Users\cloud\Documents\PowerShell\nks.omp.json
[TRACE] 13:29:41.439 shell.go:Getenv(POSH_THEME) - 0s
[DEBUG] 13:29:41.440 shell.go:Shell:638 → no shell name provided in flags, trying to detect it
[DEBUG] 13:29:41.447 shell.go:Shell:646 → process name: pwsh.exe
[TRACE] 13:29:41.447 shell.go:Shell() - 7.3327ms
[TRACE] 13:29:41.447 shell.go:resolveConfigPath() - 7.3327ms
[TRACE] 13:29:41.447 shell.go:Init() - 9.121ms
[TRACE] 13:29:41.447 shell.go:Flags() - 0s
[TRACE] 13:29:41.448 config.go:loadConfig() - 1.0191ms
[TRACE] 13:29:41.448 shell.go:Flags() - 0s
[DEBUG] 13:29:41.448 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.448 shell.go:Getenv(OMP_CACHE_DISABLED) - 70.9µs
[TRACE] 13:29:41.448 shell_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.448 shell_windows.go:WindowsRegistryKeyValue:206 → ColorizationColor(DWORD): 0xC4D0000C
[TRACE] 13:29:41.448 shell.go:Shell() - 0s
[DEBUG] 13:29:41.448 shell.go:Getenv:398 → 7.3.6
[TRACE] 13:29:41.448 shell.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 13:29:41.448 debug.go:PrintDebug:23 → Segment: Title
[TRACE] 13:29:41.448 shell_windows.go:Root() - 0s
[TRACE] 13:29:41.448 shell.go:Shell() - 0s
[TRACE] 13:29:41.448 shell.go:StatusCodes() - 0s
[TRACE] 13:29:41.448 shell_windows.go:IsWsl() - 0s
[DEBUG] 13:29:41.448 shell.go:TemplateCache:816 → environment: [ALLUSERSPROFILE=C:\ProgramData APPDATA=C:\Users\cloud\AppData\Roaming CommonProgramFiles=C:\Program Files\Common Files CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files CommonProgramW6432=C:\Program Files\Common Files COMPUTERNAME=GENESIS-NKS ComSpec=C:\Windows\system32\cmd.exe CONDA_PROMPT_MODIFIER=False DriverData=C:\Windows\System32\Drivers\DriverData GIT_INSTALL_ROOT=C:\Scoop\apps\git\current HOMEDRIVE=C: HOMEPATH=\Users\cloud IGCCSVC_DB=AQAAANCMnd8BFdERjHoAwE/Cl+sBAAAAFsU4eJG0VU+Ym8qnDmFEKAQAAAACAAAAAAAQZgAAAAEAACAAAAAgzNiQX0VtPJyF+2Shk0MfPn0t57EbRcMStrDyfk42JwAAAAAOgAAAAAIAACAAAAARp2a64Wcy3FQOUk2xOqWxtGGzcDz0VOdSLiL/fx6lIGAAAADif9MdgSfL7yk3Um33yKfCtUbwOnR1ayO+SIBzjInkoJaBKlnPTcgOFBNGPlBlm6/SsbgR2mKTS9Cgvn1PlzvnuQxv5cSkIlPLpkwJKyt5X4/uF1p4Rn8K24Sl0dJpgvdAAAAAeioarMhWAmWG/an4j9Tb2x5t5Hh7QYIeHSrSDbqP6C7kUUB0HnZ2JJi70BIHrFW5fO7sPKuPiWg3HjWjip1V8w== JAVA_HOME=C:\Program Files\Eclipse Adoptium\jdk-17.0.7.7-hotspot\ LOCALAPPDATA=C:\Users\cloud\AppData\Local LOGONSERVER=\\GENESIS-NKS MAGICK_CODER_MODULE_PATH=C:\Scoop\apps\imagemagick\current\modules\coders MAGICK_CONFIGURE_PATH=C:\Scoop\apps\imagemagick\current MAGICK_HOME=C:\Scoop\apps\imagemagick\current NUMBER_OF_PROCESSORS=20 OneDrive=C:\Users\cloud\OneDrive OneDriveConsumer=C:\Users\cloud\OneDrive OPENSSL_CONF=C:\Scoop\apps\openssl\current\bin\cnf\openssl.cnf OPENSSL_INCLUDE_DIR=C:\Scoop\apps\openssl\current\include OPENSSL_LIB_DIR=C:\Scoop\apps\openssl\current\lib OPENSSL_MODULES=C:\Scoop\apps\openssl\current\bin OS=Windows_NT Path=C:\Program Files\PowerShell\7;C:\Program Files\Docker\Docker\resources\bin;C:\Program Files (x86)\VMware\VMware Workstation\bin\;C:\Program Files\Volta\;C:\Program Files\Eclipse Adoptium\jdk-17.0.7.7-hotspot\bin;C:\Program Files\Eclipse Adoptium\jdk-11.0.19.7-hotspot\bin;C:\Program Files\Eclipse Adoptium\jdk-8.0.372.7-hotspot\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Windows\System32\OpenSSH\;C:\Program Files\PowerShell\7\;C:\Scoop\apps\imagemagick\current;C:\Users\cloud\.console-ninja\.bin;C:\Users\cloud\AppData\Local\Volta\bin;C:\Users\cloud\AppData\Local\Programs\oh-my-posh\bin;C:\Users\cloud\AppData\Local\Programs\Microsoft VS Code\bin;C:\Scoop\apps\openssl\current\bin;C:\Scoop\shims;C:\Users\cloud\AppData\Local\Microsoft\WindowsApps PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL PHP_INI_SCAN_DIR=C:\Scoop\apps\php74\current\cli;C:\Scoop\apps\php74\current\cli\conf.d; POSH_AZURE_ENABLED=False POSH_CURSOR_COLUMN=1 POSH_CURSOR_LINE=22 POSH_GIT_ENABLED=False POSH_INSTALLER=winget POSH_PID=10228 POSH_SHELL_VERSION=7.3.6 POSH_THEME=C:\Users\cloud\Documents\PowerShell\nks.omp.json POSH_THEMES_PATH=C:\Users\cloud\AppData\Local\Programs\oh-my-posh\themes POWERLINE_COMMAND=oh-my-posh POWERSHELL_DISTRIBUTION_CHANNEL=MSI:Windows 10 Pro PROCESSOR_ARCHITECTURE=AMD64 PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 154 Stepping 3, GenuineIntel PROCESSOR_LEVEL=6 PROCESSOR_REVISION=9a03 ProgramData=C:\ProgramData ProgramFiles=C:\Program Files ProgramFiles(x86)=C:\Program Files (x86) ProgramW6432=C:\Program Files PSModulePath=C:\Users\cloud\Documents\PowerShell\Modules;C:\Program Files\PowerShell\Modules;c:\program files\powershell\7\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules PUBLIC=C:\Users\Public SystemDrive=C: SystemRoot=C:\Windows TEMP=C:\Users\cloud\AppData\Local\Temp TMP=C:\Users\cloud\AppData\Local\Temp USERDOMAIN=GENESIS-NKS USERDOMAIN_ROAMINGPROFILE=GENESIS-NKS USERNAME=cloud USERPROFILE=C:\Users\cloud windir=C:\Windows WSLENV=WT_SESSION::WT_PROFILE_ID WT_PROFILE_ID={574e775e-4f2a-5b96-ac1e-a2962a402336} WT_SESSION=802843b3-f560-431e-8d38-c8ec1e04a6ee ZES_ENABLE_SYSMAN=1]
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.448 shell.go:Pwd:428 → C:\Users\cloud
[TRACE] 13:29:41.448 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.448 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[TRACE] 13:29:41.448 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.448 shell.go:User:549 → cloud
[TRACE] 13:29:41.448 shell.go:User() - 0s
[DEBUG] 13:29:41.449 shell.go:Host:561 → GENESIS-NKS
[TRACE] 13:29:41.449 shell.go:Host() - 597.8µs
[TRACE] 13:29:41.449 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.449 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.449 shell.go:Getenv(SHLVL) - 0s
[TRACE] 13:29:41.449 shell.go:TemplateCache() - 597.8µs
[TRACE] 13:29:41.449 shell.go:Flags() - 0s
[DEBUG] 13:29:41.449 block.go:Debug:280 → Segment: os
[TRACE] 13:29:41.449 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.449 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.449 shell.go:GOOS() - 0s
[TRACE] 13:29:41.449 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.449 properties.go:GetString:28 → 
[TRACE] 13:29:41.449 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.450 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.450 shell.go:Shell() - 0s
[DEBUG] 13:29:41.450 block.go:Debug:280 → Segment: session
[TRACE] 13:29:41.450 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.450 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.450 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.450 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.450 shell.go:Getenv(SSH_CONNECTION) - 0s
[DEBUG] 13:29:41.450 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.450 shell.go:Getenv(SSH_CLIENT) - 0s
[TRACE] 13:29:41.450 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.451 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.451 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.451 shell.go:Shell() - 0s
[DEBUG] 13:29:41.451 block.go:Debug:280 → Segment: shell
[TRACE] 13:29:41.451 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.451 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.451 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.451 properties.go:GetKeyValueMap:46 → mapped_shell_names: map[bash: bash pwsh: pwsh]
[TRACE] 13:29:41.451 shell.go:Shell() - 0s
[TRACE] 13:29:41.451 shell.go:Flags() - 0s
[TRACE] 13:29:41.451 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.451 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.451 shell.go:Shell() - 0s
[DEBUG] 13:29:41.451 block.go:Debug:280 → Segment: sysinfo
[DEBUG] 13:29:41.451 properties.go:GetInt:40 → precision: 2
[TRACE] 13:29:41.451 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.451 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.451 shell.go:GOOS() - 0s
[TRACE] 13:29:41.451 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.451 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.451 shell.go:Shell() - 0s
[DEBUG] 13:29:41.451 block.go:Debug:280 → Segment: sysinfo
[DEBUG] 13:29:41.451 properties.go:GetInt:40 → precision: 2
[TRACE] 13:29:41.451 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.451 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.451 shell.go:GOOS() - 0s
[TRACE] 13:29:41.451 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.451 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.451 shell.go:Shell() - 0s
[DEBUG] 13:29:41.451 block.go:Debug:280 → Segment: executiontime
[TRACE] 13:29:41.451 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.451 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.451 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.451 properties.go:GetBool:22 → always_enabled: false
[TRACE] 13:29:41.451 shell.go:ExecutionTime() - 0s
[DEBUG] 13:29:41.451 properties.go:GetFloat64:34 → threshold: 500.000000
[TRACE] 13:29:41.451 shell.go:Flags() - 0s
[DEBUG] 13:29:41.451 block.go:Debug:280 → Segment: git
[TRACE] 13:29:41.451 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.451 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.451 shell.go:GOOS() - 0s
[TRACE] 13:29:41.451 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.456 shell.go:CommandPath:600 → C:\Scoop\shims\git.exe
[TRACE] 13:29:41.456 shell.go:CommandPath(git.exe) - 4.8869ms
[TRACE] 13:29:41.456 shell.go:HasCommand(git.exe) - 4.8869ms
[TRACE] 13:29:41.456 shell.go:Pwd() - 0s
[ERROR] 13:29:41.457 shell.go:HasParentFilePath:734 → CreateFile .git: The system cannot find the file specified.
[TRACE] 13:29:41.457 shell.go:HasParentFilePath(.git) - 539.8µs
[DEBUG] 13:29:41.457 properties.go:GetBool:22 → fetch_bare_info: false
[DEBUG] 13:29:41.457 block.go:Debug:280 → Segment: text
[TRACE] 13:29:41.457 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.457 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.457 shell.go:GOOS() - 0s
[TRACE] 13:29:41.457 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.457 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.457 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.457 shell.go:Shell() - 0s
[DEBUG] 13:29:41.457 block.go:Debug:280 → Segment: python
[DEBUG] 13:29:41.457 properties.go:GetString:28 → environment
[TRACE] 13:29:41.457 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.457 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.457 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.457 properties.go:GetStringArray:52 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 13:29:41.457 properties.go:GetStringArray:52 → folders: [.venv venv virtualenv env venv-win pyenv-win]
[DEBUG] 13:29:41.457 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:29:41.457 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.457 shell_windows.go:56 → C:\Users\cloud
[DEBUG] 13:29:41.457 block.go:Debug:280 → Segment: node
[TRACE] 13:29:41.457 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.457 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.457 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.457 properties.go:GetStringArray:52 → extensions: [*.js *.ts package.json .nvmrc pnpm-workspace.yaml .pnpmfile.cjs .npmrc .vue]
[DEBUG] 13:29:41.457 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:29:41.457 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:29:41.457 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.457 shell_windows.go:56 → C:\Users\cloud
[DEBUG] 13:29:41.457 block.go:Debug:280 → Segment: java
[DEBUG] 13:29:41.457 shell.go:Getenv:398 → C:\Program Files\Eclipse Adoptium\jdk-17.0.7.7-hotspot\
[TRACE] 13:29:41.457 shell.go:Getenv(JAVA_HOME) - 0s
[TRACE] 13:29:41.457 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.457 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.457 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.457 properties.go:GetStringArray:52 → extensions: [pom.xml build.gradle.kts build.sbt .java-version .deps.edn project.clj build.boot *.java *.class *.gradle *.jar *.clj *.cljc]
[DEBUG] 13:29:41.457 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:29:41.457 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:29:41.457 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.457 shell_windows.go:56 → C:\Users\cloud
[DEBUG] 13:29:41.457 block.go:Debug:280 → Segment: php
[TRACE] 13:29:41.457 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.457 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.457 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.457 properties.go:GetStringArray:52 → extensions: [*.php composer.json composer.lock .php-version blade.php]
[DEBUG] 13:29:41.457 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:29:41.457 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:29:41.457 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.457 shell_windows.go:56 → C:\Users\cloud
[DEBUG] 13:29:41.457 block.go:Debug:280 → Segment: text
[TRACE] 13:29:41.457 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.457 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.457 shell.go:GOOS() - 0s
[TRACE] 13:29:41.457 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.457 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.457 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.457 shell.go:Shell() - 0s
[DEBUG] 13:29:41.457 block.go:Debug:280 → Segment: battery
[TRACE] 13:29:41.457 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.457 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.457 shell.go:GOOS() - 0s
[TRACE] 13:29:41.457 shell_windows.go:IsWsl() - 0s
[TRACE] 13:29:41.464 shell_windows_nix.go:BatteryState() - 6.497ms
[DEBUG] 13:29:41.464 properties.go:GetString:28 → 
[TRACE] 13:29:41.464 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.464 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.464 shell.go:Shell() - 0s
[TRACE] 13:29:41.464 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.465 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.465 shell.go:TemplateCache() - 0s
[DEBUG] 13:29:41.465 block.go:Debug:280 → Segment: connection
[TRACE] 13:29:41.465 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.465 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.465 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.465 properties.go:GetString:28 → wifi|ethernet
[TRACE] 13:29:41.466 networks_windows.go:wifiNetwork() - 0s
[TRACE] 13:29:41.468 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.468 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.468 shell.go:Shell() - 0s
[TRACE] 13:29:41.469 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.469 shell.go:Flags() - 0s
[DEBUG] 13:29:41.469 block.go:Debug:280 → Segment: kubectl
[TRACE] 13:29:41.469 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.469 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.469 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.469 properties.go:GetBool:22 → parse_kubeconfig: false
[DEBUG] 13:29:41.469 shell.go:CommandPath:600 → C:\Program Files\Docker\Docker\resources\bin\kubectl.exe
[TRACE] 13:29:41.469 shell.go:CommandPath(kubectl) - 520.4µs
[TRACE] 13:29:41.469 shell.go:HasCommand(kubectl) - 520.4µs
[ERROR] 13:29:41.533 shell.go:RunCommand:577 → exit status 1
[DEBUG] 13:29:41.533 shell.go:RunCommand:579 ↓
    error: current-context must exist in order to minify

[TRACE] 13:29:41.533 shell.go:RunCommand(kubectl config view --output yaml --minify) - 63.6545ms
[DEBUG] 13:29:41.533 properties.go:GetBool:22 → display_error: false
[DEBUG] 13:29:41.533 block.go:Debug:280 → Segment: aws
[TRACE] 13:29:41.533 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.533 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.533 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.533 properties.go:GetBool:22 → display_default: true
[DEBUG] 13:29:41.533 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.533 shell.go:Getenv(AWS_VAULT) - 0s
[DEBUG] 13:29:41.533 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.533 shell.go:Getenv(AWS_DEFAULT_PROFILE) - 0s
[DEBUG] 13:29:41.533 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.533 shell.go:Getenv(AWS_PROFILE) - 0s
[DEBUG] 13:29:41.533 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.533 shell.go:Getenv(AWS_REGION) - 0s
[DEBUG] 13:29:41.533 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.533 shell.go:Getenv(AWS_DEFAULT_REGION) - 0s
[DEBUG] 13:29:41.533 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.533 shell.go:Getenv(AWS_CONFIG_FILE) - 0s
[DEBUG] 13:29:41.533 shell_windows.go:56 → C:\Users\cloud
[ERROR] 13:29:41.533 shell.go:FileContent:519 → open C:\Users\cloud/.aws/config: The system cannot find the path specified.
[TRACE] 13:29:41.533 shell.go:FileContent(C:\Users\cloud/.aws/config) - 0s
[DEBUG] 13:29:41.533 block.go:Debug:280 → Segment: az
[TRACE] 13:29:41.533 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.533 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.533 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.533 properties.go:GetString:28 → first_match
[DEBUG] 13:29:41.533 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.533 shell.go:Getenv(AZURE_CONFIG_DIR) - 0s
[DEBUG] 13:29:41.533 shell_windows.go:56 → C:\Users\cloud
[DEBUG] 13:29:41.533 shell_windows.go:56 → C:\Users\cloud
[DEBUG] 13:29:41.533 shell.go:HasFilesInDir:464 → false
[TRACE] 13:29:41.533 shell.go:HasFilesInDir(azureProfile.json) - 0s
[DEBUG] 13:29:41.533 shell.go:HasFilesInDir:464 → false
[TRACE] 13:29:41.533 shell.go:HasFilesInDir(azureProfile.json) - 0s
[DEBUG] 13:29:41.533 shell.go:Getenv:398 → NO DATA
[TRACE] 13:29:41.533 shell.go:Getenv(POSH_AZURE_SUBSCRIPTION) - 0s
[DEBUG] 13:29:41.533 block.go:Debug:280 → Segment: cftarget
[TRACE] 13:29:41.533 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.533 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.533 shell.go:GOOS() - 0s
[ERROR] 13:29:41.538 shell.go:CommandPath:603 → exec: "cf": executable file not found in %PATH%
[TRACE] 13:29:41.538 shell.go:CommandPath(cf) - 5.3821ms
[TRACE] 13:29:41.538 shell.go:HasCommand(cf) - 5.3821ms
[TRACE] 13:29:41.538 shell.go:Flags() - 0s
[DEBUG] 13:29:41.538 block.go:Debug:280 → Segment: text
[TRACE] 13:29:41.538 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.538 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.538 shell.go:Shell() - 0s
[DEBUG] 13:29:41.538 block.go:Debug:280 → Segment: time
[TRACE] 13:29:41.538 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.538 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.538 properties.go:GetString:28 → 15:04
[TRACE] 13:29:41.538 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.538 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.538 shell.go:Shell() - 0s
[DEBUG] 13:29:41.538 block.go:Debug:280 → Segment: root
[TRACE] 13:29:41.538 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.538 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell_windows.go:Root() - 0s
[DEBUG] 13:29:41.538 block.go:Debug:280 → Segment: path
[TRACE] 13:29:41.538 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.538 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:Pwd() - 0s
[TRACE] 13:29:41.538 shell.go:Shell() - 0s
[TRACE] 13:29:41.538 shell.go:Flags() - 0s
[DEBUG] 13:29:41.538 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 13:29:41.538 properties.go:GetString:28 → 
[DEBUG] 13:29:41.538 properties.go:GetString:28 → 
[DEBUG] 13:29:41.538 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[DEBUG] 13:29:41.538 properties.go:GetString:28 → 󰋜
[DEBUG] 13:29:41.538 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 13:29:41.538 shell.go:PathSeparator() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:PathSeparator() - 0s
[TRACE] 13:29:41.538 shell.go:PathSeparator() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:PathSeparator() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:PathSeparator() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:GOOS() - 0s
[TRACE] 13:29:41.538 shell.go:PathSeparator() - 0s
[TRACE] 13:29:41.538 shell.go:Pwd() - 0s
[TRACE] 13:29:41.538 shell_windows.go:IsWsl() - 0s
[TRACE] 13:29:41.538 shell.go:StackCount() - 0s
[DEBUG] 13:29:41.539 win32_windows.go:isWriteable:299 → not current user or in group
[DEBUG] 13:29:41.539 win32_windows.go:isWriteable:303 → current user is member of S-1-5-32-544
[DEBUG] 13:29:41.539 win32_windows.go:isWriteable:311 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 13:29:41.539 win32_windows.go:isWriteable:313 → user has write access
[TRACE] 13:29:41.539 shell_windows.go:DirIsWritable() - 551.4µs
[TRACE] 13:29:41.539 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.539 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.540 shell.go:Shell() - 0s
[TRACE] 13:29:41.540 shell.go:Flags() - 0s
[DEBUG] 13:29:41.540 block.go:Debug:280 → Segment: exit
[DEBUG] 13:29:41.540 properties.go:GetString:28 → {{ .Code }}
[TRACE] 13:29:41.540 shell.go:Pwd() - 0s
[DEBUG] 13:29:41.540 shell_windows.go:56 → C:\Users\cloud
[TRACE] 13:29:41.540 shell.go:GOOS() - 0s
[TRACE] 13:29:41.540 shell.go:StatusCodes() - 0s
[TRACE] 13:29:41.540 shell.go:TemplateCache() - 0s
[DEBUG] 13:29:41.540 properties.go:GetBool:22 → always_enabled: true
[TRACE] 13:29:41.540 shell.go:TemplateCache() - 0s
[TRACE] 13:29:41.540 shell.go:Shell() - 0s
[DEBUG] 13:29:41.540 shell.go:Getenv:398 → C:\Users\cloud\AppData\Local
[TRACE] 13:29:41.540 shell.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 13:29:41.540 shell_windows.go:CachePath() - 0s
[TRACE] 13:29:41.540 shell.go:Flags() - 0s
```

--- a/src/platform/concurrent_map.go
+++ b/src/platform/concurrent_map.go
@@ -37,3 +37,12 @@ func (c *ConcurrentMap) Delete(key string) {
 func (c *ConcurrentMap) List() map[string]interface{} {
 	return c.values
 }
+
+func (c *ConcurrentMap) Contains(key string) bool {
+	c.RLock()
+	defer c.RUnlock()
+	if _, ok := c.values[key]; ok {
+		return true
+	}
+	return false
+}

--- a/src/platform/shell.go
+++ b/src/platform/shell.go
@@ -166,13 +166,6 @@ type SystemInfo struct {
 	Disks map[string]disk.IOCountersStat
 }
 
-type SegmentsCache map[string]interface{}
-
-func (s *SegmentsCache) Contains(key string) bool {
-	_, ok := (*s)[key]
-	return ok
-}
-
 type TemplateCache struct {
 	Root         bool
 	PWD          string
@@ -188,22 +181,18 @@ type TemplateCache struct {
 	WSL          bool
 	PromptCount  int
 	SHLVL        int
-	Segments     SegmentsCache
+	Segments     *ConcurrentMap
 
 	initialized bool
 	sync.RWMutex
 }
 
 func (t *TemplateCache) AddSegmentData(key string, value interface{}) {
-	t.Lock()
-	t.Segments[key] = value
-	t.Unlock()
+	t.Segments.Set(key, value)
 }
 
 func (t *TemplateCache) RemoveSegmentData(key string) {
-	t.Lock()
-	delete(t.Segments, key)
-	t.Unlock()
+	t.Segments.Delete(key)
 }
 
 type Environment interface {
@@ -802,7 +791,7 @@ func (env *Shell) TemplateCache() *TemplateCache {
 	tmplCache.ShellVersion = env.CmdFlags.ShellVersion
 	tmplCache.Code, _ = env.StatusCodes()
 	tmplCache.WSL = env.IsWsl()
-	tmplCache.Segments = make(map[string]interface{})
+	tmplCache.Segments = NewConcurrentMap()
 	tmplCache.PromptCount = env.CmdFlags.PromptCount
 	tmplCache.Env = make(map[string]string)
 	tmplCache.Var = make(map[string]interface{})

--- a/src/platform/shell_windows.go
+++ b/src/platform/shell_windows.go
@@ -148,9 +148,12 @@ func (env *Shell) WindowsRegistryKeyValue(path string) (*WindowsRegistryValue, e
 		return nil, err
 	}
 
-	regKey := Base(env, regPath)
-	if len(regKey) != 0 {
-		regPath = strings.TrimSuffix(regPath, `\`+regKey)
+	var regKey string
+	if !strings.HasSuffix(regPath, `\`) {
+		regKey = Base(env, regPath)
+		if len(regKey) != 0 {
+			regPath = strings.TrimSuffix(regPath, `\`+regKey)
+		}
 	}
 
 	var key registry.Key

--- a/src/template/text.go
+++ b/src/template/text.go
@@ -159,6 +159,12 @@ func (t *Text) cleanTemplate() {
 			// end of a variable, needs to be appended
 			if !isKnownVariable(property) {
 				result += ".Data" + property
+			} else if strings.HasPrefix(property, ".Segments") && !strings.HasSuffix(property, ".Contains") {
+				// as we can't provide a clean way to access the list
+				// of segments, we need to replace the property with
+				// the list of segments so they can be accessed directly
+				property = strings.Replace(property, ".Segments", ".Segments.List", 1)
+				result += property
 			} else {
 				// check if we have the same property in Data
 				// and replace it with the Data property so it

--- a/src/template/text_test.go
+++ b/src/template/text_test.go
@@ -320,6 +320,16 @@ func TestCleanTemplate(t *testing.T) {
 			Expected: "{{.Data.OS}}",
 			Template: "{{.OS}}",
 		},
+		{
+			Case:     "Keep .Contains intact for Segments",
+			Expected: `{{.Segments.Contains "Git"}}`,
+			Template: `{{.Segments.Contains "Git"}}`,
+		},
+		{
+			Case:     "Replace a direct call to .Segments with .Segments.List",
+			Expected: `{{.Segments.List.Git.Repo}}`,
+			Template: `{{.Segments.Git.Repo}}`,
+		},
 	}
 	for _, tc := range cases {
 		tmpl := &Text{
@@ -342,11 +352,11 @@ func TestSegmentContains(t *testing.T) {
 	}
 
 	env := &mock.MockedEnvironment{}
+	segments := platform.NewConcurrentMap()
+	segments.Set("Git", "foo")
 	env.On("TemplateCache").Return(&platform.TemplateCache{
-		Env: make(map[string]string),
-		Segments: map[string]interface{}{
-			"Git": nil,
-		},
+		Env:      make(map[string]string),
+		Segments: segments,
 	})
 	for _, tc := range cases {
 		tmpl := &Text{

